sphinx/environment.py

```diff
@@ -976,25 +1000,13 @@ def process_images(self, docname, doctree):
             # set imgpath as default URI
             node['uri'] = rel_imgpath
             if rel_imgpath.endswith(os.extsep + '*'):
-                for filename in glob(full_imgpath):
-                    new_imgpath = relative_path(path.join(self.srcdir, 'dummy'),
-                                                filename)
-                    if filename.lower().endswith('.pdf'):
-                        candidates['application/pdf'] = new_imgpath
-                    elif filename.lower().endswith('.svg'):
-                        candidates['image/svg+xml'] = new_imgpath
-                    else:
-                        try:
-                            f = open(filename, 'rb')
-                            try:
-                                imgtype = imghdr.what(f)
-                            finally:
-                                f.close()
-                        except (OSError, IOError) as err:
-                            self.warn_node('image file %s not readable: %s' %
-                                           (filename, err), node)
-                        if imgtype:
-                            candidates['image/' + imgtype] = new_imgpath
+                if self.config.figure_language:
+                    # Search language-specific figures at first
+                    i18n_imguri = get_image_filename_for_locale(imguri, self, force=True)
+                    _, full_i18n_imgpath = self.relfn2path(i18n_imguri, docname)
+                    collect_candidates(full_i18n_imgpath, candidates)
+
+                collect_candidates(full_imgpath, candidates)
```

Comment:
This process is executed on reading phase.
So, at this moment, any preferred mime types could not be determined yet, I think.
Are there any short circuits?